src/raft/raft.rs

```diff
@@ -304,6 +304,11 @@ impl<T: Storage> Raft<T> {
         hs
     }
 
+    pub fn in_lease(&self, safe_distance: usize) -> bool {
+        self.state == StateRole::Leader && self.check_quorum &&
+        self.election_elapsed + safe_distance <= self.election_timeout
```

Comment:
I think `safe-distance` here is something more like placebo before https://github.com/pingcap/tikv/issues/841 is resolved.